### DISCUSSION TITLE
[ QUESTION ] Can Additional Filters be Added to Get-FalconHost?

### DISCUSSION THREAD

**Discussion by oracle701** [CrowdStrike/psfalcon#182] [Q&A] [answered]
Hello,
I am working on a script\project and I was wondering if a filter for "grouping tags" and "group" can be added to Get-FalconHost? I need to filter and grab servers only, however that doesn't always align with the OS versions for me (ie Windows 7 but considered a Server, with a "server" tag and in a "server" group).

**Environment:**
 - OS: Windows 10
 - PowerShell: 7.2
 - PSFalcon: 2.1.7

**Comment by bk-cs** [comment, marked as answer]
The values that are used within the `-Filter` parameter are defined by the API and the [Falcon Query Language](https://github.com/CrowdStrike/psfalcon/wiki/Falcon-Query-Language). 

For `Get-FalconHost`, I believe you can use `tags` and `groups`:
```
Get-FalconHost -Filter "tags:'SensorGroupingTags/Example'+groups:'<group_id>'"
```

**Comment by oracle701** [comment]
I did try using just "tags" like that, but it does not work. 
Same result when I try "tag", and "groups" or "group" too.

Command:
Get-FalconHost -Filter "tags:'SensorGroupingTags/MyTag'" -Detailed -All

Result:
Get-FalconHost: Cannot validate argument on parameter 'Filter'. 'tags' is not a valid property.

**Reply by bk-cs** [reply]
This is a bug that is specific to your version of PSFalcon. If you upgrade to v2.1.7, it should resolve that.

You can update with:
```
Update-Module -Name PSFalcon
```
Once complete, you'll need to restart your PowerShell session and re-import the module.

If you run into any problems doing it that way, you can also uninstall, re-install, restart PowerShell and re-import:
```
Uninstall-Module -Name PSFalcon -AllVersions
Install-Module -Name PSFalcon
```

**Comment by oracle701** [comment]
Well I was on v2.1.7, so I removed it all and reinstalled. Working as expected now, thank you!
